Automattic/simplenote-ios
# Search: Wiring Results

### Details:
This PR is part of a multi-step implementation of the new Advanced Search mechanism. 

- **SPResultsController:** Encapsulates the Search Result's FRC.
- **SPNoteTableViewCell:** New a `cellHeight` static API
- **SPSearchResultsViewController:** now renders actual live results

I just... hope the names aren't confusing. Suggestions always welcome!!!

@aerych Eric!!! May I bug you with this one?
Thanks in advance!!

Ref. #507

## Known Limitations:
- SearchResults bottom padding isn't accounting for the keyboard. Addressed via #538 

### Test: Note Cells UX
0. Set the `FeatureManager.advancedSearchEnabled` property to `true`
1. Log into your account
2. Tap over the **Search** field

- [x] Verify **all of your notes** show up immediately
- [x] Verify the Search Results cell look great!

### Test: Search Results
0. Set the `FeatureManager.advancedSearchEnabled` property to `true`
1. Log into your account
2. Tap over the **Search** field
3. Enter any keyword

- [x] Verify the related notes show up
- [x] Verify the Sort Options match the Notes List sort options
- [x] Verify that pressing over any results cell will push the Editor UI

### Test: NavigationBar's Blur
0. Set the `FeatureManager.advancedSearchEnabled` property to `true`
1. Log into your account
2. Scroll down so that the NavigationBar Blur is activated
3. Tap over the **Search** field

- [x] Verify that the NavigationBar's Blur remains active, and there is no flicker / anything weird!

### Test: Swipe Back
0. Set the `FeatureManager.advancedSearchEnabled` property to `true`
1. Log into your account
2. Tap over the **Search** field
3. Open any of the Search Results

- [x] Verify that you can swipe back from the left edge of the screen

### Release
These changes do not require release notes.
